activiti-cloud-runtime-bundle-service/activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/src/main/java/org/activiti/services/connectors/channel/ServiceTaskIntegrationErrorEventHandler.java

```diff
@@ -0,0 +1,112 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ServiceTaskIntegrationErrorEventHandler.class);
+
+    private final RuntimeService runtimeService;
+    private final IntegrationContextService integrationContextService;
+    private final MessageChannel auditProducer;
+    private final RuntimeBundleProperties runtimeBundleProperties;
+    private final RuntimeBundleInfoAppender runtimeBundleInfoAppender;
+
+    public ServiceTaskIntegrationErrorEventHandler(RuntimeService runtimeService,
+                                                    IntegrationContextService integrationContextService,
+                                                    MessageChannel auditProducer,
+                                                    RuntimeBundleProperties runtimeBundleProperties,
+                                                    RuntimeBundleInfoAppender runtimeBundleInfoAppender) {
+        this.runtimeService = runtimeService;
+        this.integrationContextService = integrationContextService;
+        this.auditProducer = auditProducer;
+        this.runtimeBundleProperties = runtimeBundleProperties;
+        this.runtimeBundleInfoAppender = runtimeBundleInfoAppender;
+    }
+
+    @StreamListener(ProcessEngineIntegrationChannels.INTEGRATION_ERRORS_CONSUMER)
+    public void receive(IntegrationError integrationError) {
+        IntegrationContext integrationContext = integrationError.getIntegrationContext();
+        IntegrationContextEntity integrationContextEntity = integrationContextService.findById(integrationContext.getId());
+
+        if (integrationContextEntity != null) {
+            integrationContextService.deleteIntegrationContext(integrationContextEntity);
+
+            List<Execution> executions = runtimeService.createExecutionQuery().executionId(integrationContextEntity.getExecutionId()).list();
+            if (executions.size() > 0) {
+                // TODO map error to integrationContextEntity.getExecutionId()
+                String message = "Received integration error with execution id `" +
+                        integrationContextEntity.getExecutionId() +
+                        ", flow node id `" + integrationContext.getClientId() +
+                        "`. The integration error for the integration context `" + integrationContext.getId() + "` is {}";
+
+                LOGGER.debug(message, integrationError);
+
+            } else {
+                String message = "No task is in this RB is waiting for integration result with execution id `" +
+                    integrationContextEntity.getExecutionId() +
+                    ", flow node id `" + integrationContext.getClientId() +
+                    "`. The integration result for the integration context `" + integrationContext.getId() + "` will be ignored.";
+                LOGGER.debug(message);
+            }
+
+            sendAuditMessage(integrationError);
+        }
+    }
+
+    @SuppressWarnings("rawtypes")
```

Comment:
This can be removed if you you change `106` to `Message<CloudRuntimeEvent<?,?>[]`.

Reply:
Fixed